pkg/registry/registry_to_model.go

```diff
@@ -0,0 +1,241 @@
+			var v property.GVK
+			if err := json.Unmarshal(json.RawMessage(p.Value), &v); err != nil {
+				return nil, property.ParseError{Idx: i, Typ: p.Type, Err: err}
+			}
+			k := property.GVKProvided{Group: v.Group, Kind: v.Kind, Version: v.Version}
+			providedGVKs[k] = &property.GVKProvided{Group: v.Group, Kind: v.Kind, Version: v.Version, Plural: v.Plural}
+		case property.TypePackage:
+			foundPackageProperty = true
+			out = append(out, property.Property{
+				Type:  property.TypePackageProvided,
+				Value: json.RawMessage(p.Value),
+			})
+			out = append(out, property.Property{
+				Type:  property.TypePackage,
+				Value: json.RawMessage(p.Value),
+			})
+		default:
+			out = append(out, property.Property{
+				Type:  p.Type,
+				Value: json.RawMessage(p.Value),
+			})
+		}
+	}
+
+	for i, p := range b.Dependencies {
+		switch p.Type {
+		case property.TypeGVK:
+			var v property.GVK
+			if err := json.Unmarshal(json.RawMessage(p.Value), &v); err != nil {
+				return nil, property.ParseError{Idx: i, Typ: p.Type, Err: err}
+			}
+			k := property.GVKRequired{Group: v.Group, Kind: v.Kind, Version: v.Version}
+			requiredGVKs[k] = &property.GVKRequired{Group: v.Group, Kind: v.Kind, Version: v.Version, Plural: v.Plural}
+		case property.TypePackage:
+			out = append(out, property.Property{
+				Type:  property.TypePackageRequired,
+				Value: json.RawMessage(p.Value),
+			})
+		}
+	}
+
+	if !foundPackageProperty {
+		out = append(out, property.MustBuildPackage(b.Annotations.PackageName, version))
+		out = append(out, property.MustBuildPackageProvided(b.Annotations.PackageName, version))
+	}
+	providedApis, err := b.ProvidedAPIs()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get Provided APIs from bundle:%s", err)
+	}
+	for p, _ := range providedApis {
+		k := property.GVKProvided{Group: p.Group, Kind: p.Kind, Version: p.Version, Plural: p.Plural}
+		if v, ok := providedGVKs[k]; !ok {
+			providedGVKs[k] = &property.GVKProvided{Group: p.Group, Kind: p.Kind, Version: p.Version, Plural: p.Plural}
+		} else {
+			v.Plural = p.Plural
+		}
+	}
+	requiredApis, err := b.RequiredAPIs()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get Required APIs from bundle:%s", err)
+	}
+	for p, _ := range requiredApis {
+		k := property.GVKRequired{Group: p.Group, Kind: p.Kind, Version: p.Version}
+		if v, ok := requiredGVKs[k]; !ok {
+			requiredGVKs[k] = &property.GVKRequired{Group: p.Group, Kind: p.Kind, Version: p.Version, Plural: p.Plural}
+		} else {
+			v.Plural = p.Plural
+		}
+	}
+
+	for _, p := range providedGVKs {
+		out = append(out, property.MustBuildGVKProvided(p.Group, p.Version, p.Kind, p.Plural))
+		out = append(out, property.MustBuildGVK(p.Group, p.Version, p.Kind, ""))
+	}
+
+	for _, p := range requiredGVKs {
+		out = append(out, property.MustBuildGVKRequired(p.Group, p.Version, p.Kind, p.Plural))
+	}
+
+	return out, nil
+}
+
+func convertToModelRelatedImages(csv *ClusterServiceVersion) ([]model.RelatedImage, error) {
```

Comment:
I have a similar implementation here: https://github.com/joelanford/operator-registry/blob/f81871a10201ea7e096f75361701f12fc53220c1/pkg/api/api_to_model.go#L128-L149

If we can find a place to centralize, perhaps we should?

Reply:
Disregard. We talked about how this code is necessary for now because we want conversion functions from the various types to also go into the model.

Unfortunately, we can't re-use the api conversion method because it takes `api.Bundle`, not `registry.Bundle`.

In the future, perhaps we can look at converting a bundle image directly to `model.Bundle`